r/src/recordbatchreader.cpp

```diff
@@ -74,6 +76,12 @@ int ipc___RecordBatchFileReader__num_record_batches(
   return reader->num_record_batches();
 }
 
+// [[arrow::export]]
+cpp11::strings ipc___RecordBatchFileReader__metadata(
```

Comment:
It's cool that you added this but it's probably not necessary; see below for a simpler test.

Reply:
I'll revert this, then